Automattic/jetpack
# Social | Add site context for publicize endpoints

## Proposed changes:

* Add site context to `publicize/connections` endpoint to allow fetching of site-wide connections.

The implementation is that in order to get site-wide connections, the request should be sent as a blog instead of a user.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Sandbox this PR
* As an admin on a Jetpack site add some social connections.
* Add some connections as a different author
* As the author, goto editor
* Open browser console and enter this
```js
await wp.apiFetch({
    path: '/wpcom/v2/publicize/connections'
})
```
* Verify that it returns the connections for the current user by default
* Run the same when logged in as admin
* Verify that it returns all the connections for the site
* Mark a connection or two as shared from admin account
* Run this as author account to get the shared connections as well
```js
await wp.apiFetch({
    path: '/wpcom/v2/publicize/connections'
})
```
* Verify that the request is successful and includes shared connections
* Verify that running the request from [developer console](https://developer.wordpress.com/docs/api/console/) returns error.
